trusted/crypto/src/lib.rs

```diff
@@ -37,32 +56,137 @@ impl Sha256 {
     }
 }
 
+/// Builds a [Sha256] hash in a piecewise fashion.  This can be used when all
+/// of the data to hash is not available in contiguous memory.
+pub struct Sha256Builder {
+    handle: sgx_sha_state_handle_t,
+}
+
+impl Sha256Builder {
```

Comment:
Any thoughts on this name?  I went with "builder" because of the builder pattern but wasn't sure if something like "hasher" would make more sense.

cc: @remoun since you already looked over this once.